### DISCUSSION TITLE
Array and Structs

### DISCUSSION THREAD

**Discussion by shrivrat0809** [Cyfrin/foundry-full-course-cu#328] [Q&A] [answered]
just like in the tutorial is it necessary to create a struct before a array to make list. Like first create a struct than create variable then use function to do further things ? 
HELP ME I AM A LITTLE BIT CONFUSED BETWEEN THESE TWO.

**Comment by shikhar229169** [comment, marked as answer]
Actually it is not necessary to create a struct before an array. But we are following a pattern to write our code, to make it
- Easy for others to read our code.
- Easy to debug code, and locate all the variables, functions, structs easily.
- Appears more structured.

It is considered as a good practice to arrange the things in a particular order, as described in the course.

**Reply by cromewar** [reply]
Amazing anser @shikhar229169
